### DISCUSSION TITLE
Raw result & RetrieveMultiple plugin

### DISCUSSION THREAD

**Discussion by chironh** [rappen/FetchXMLBuilder#725] [Q&A]
Having a RetrieveMultiple plugin on a specific entity, "Raw result" doesn't show results whereas all other formats (JSON, XML, View) show the correct output.

If I throw purposely an exception in the RetrieveMultiple, all other formats give me the desired exception (which means the RetrieveMultiple has been triggered), except for the "Raw result" which just shows me

`<resultset morerecords="0" />`

What's the difference between executing a fetch as "Raw result" and the others? What kind of query does it uses behind the scenes?

**Comment by rappen** [comment]
Interesting!
Do you have the query to share?

Read more:
Raw uses the internal `ExecuteFetch` method, the others use internal `RetrieveMultiple` method. [see the code](https://github.com/rappen/FetchXMLBuilder/blob/30d3bf780d8352c9b6f95804ed85b76f3a1da30d/FetchXmlBuilder/FXBQueries.cs#L82-L99)
The `ExecuteFetch` calls a `Service.Execute(new ExecuteFetchRequest { FetchXml = fetch })` [see code](https://github.com/rappen/FetchXMLBuilder/blob/30d3bf780d8352c9b6f95804ed85b76f3a1da30d/FetchXmlBuilder/FXBQueries.cs#L220)
The `RetrieveMultiple` calls a `Service.RetrieveMultiple(new FetchExpression(fetch))` [see code](https://github.com/rappen/FetchXMLBuilder/blob/30d3bf780d8352c9b6f95804ed85b76f3a1da30d/FetchXmlBuilder/FXBQueries.cs#L268)

**Comment by chironh** [comment]
Ah, that would explain. Another message, hence why the RetrieveMultiple was never triggered. Good to know :)

Maybe on a (small) sidenote: ExecuteFetchRequest is actually deprecated, but apparantly it still works :) But that also means the "Raw result" is not the preferred way to go.

https://docs.microsoft.com/en-us/dotnet/api/microsoft.crm.sdk.messages.executefetchrequest?view=dataverse-sdk-latest

`Deprecated. Use the [RetrieveMultipleRequest](https://docs.microsoft.com/en-us/dotnet/api/microsoft.xrm.sdk.messages.retrievemultiplerequest?view=dataverse-sdk-latest) class.`

**Comment by rappen** [comment]
Thanks for that help - created a new issue #726

**Comment by chironh** [comment]
Keep up the awesome work *thumbsup*
